lib/server/services/dpl/getOrCreateDplDetector.js

```diff
@@ -11,12 +11,44 @@
  *  or submit itself to any jurisdiction.
  */
 
-const { DplDetectorRepository } = require('../../../database/repositories/index.js');
+const { sequelize } = require('../../../database/index.js');
+const { DplDetectorRepository, DetectorRepository } = require('../../../database/repositories/index.js');
+const { DetectorType } = require('../../../domain/enums/DetectorTypes.js');
 
 /**
  * Fetch (and create if it does not exist) a DPL detector
  *
  * @param {string} name the name of the dpl detector to find/create
  * @return {Promise<SequelizeDplDetector>} the dpl detector found
  */
-exports.getOrCreateDplDetectorByName = (name) => DplDetectorRepository.findOneOrCreate({ name });
+exports.getOrCreateDplDetectorByName = async (name) => {
+    let dplDetector = await DplDetectorRepository.findOne({ where: { name } });
+    if (dplDetector) {
+        return dplDetector;
+    } else {
```

Comment:
No need for this else, you can simply continue.

Reply:
applied